File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m - 1, j = n - 1, ins = m + n - 1;
        while (i >= 0 || j >= 0) {
            if (i == -1) {
                nums1[ins] = nums2[j];
                j--;
            } else if (j == -1 || nums1[i] > nums2[j]) {
                nums1[ins] = nums1[i];
                i--;
            } else {
                nums1[ins] = nums2[j];
                j--;
            }
            ins--;
        }
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.swift
class Solution {
    func merge(_ nums1: inout [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        var i = 0, j = 0
        var rebuilt = [Int]()
        while i < m || j < n {
            if i == m {
                rebuilt.append(nums2[j])
                j += 1
            } else if j == n || nums1[i] < nums2[j] {
                rebuilt.append(nums1[i])
                i += 1
            } else {
                rebuilt.append(nums2[j])
                j += 1
            }

        }
        nums1 = rebuilt
    }
}